main.tf

```diff
@@ -13,6 +13,65 @@ module "label" {
   tags       = "${var.tags}"
 }
 
+resource "aws_s3_bucket" "cache_bucket" {
+  count         = "${var.cache_enabled == "true" ? 1 : 0}"
+  bucket        = "${local.cache_bucket_name_normalised}"
+  acl           = "private"
+  force_destroy = true
+  tags          = "${module.label.tags}"
+
+  lifecycle_rule {
+    id      = "codebuildcache"
+    enabled = true
+
+    prefix = "/"
+    tags   = "${module.label.tags}"
+
+    expiration {
+      days = "${var.cache_expiration_days}"
+    }
+  }
+}
+
+provider "random" {
+  version = "~> 1.0"
+}
+
+provider "null" {
+  version = "~> 1.0"
+}
```

Comment:
why do we need to specify these two providers here?

Reply:
I've added them here because users using the module may not know that the warnings about the provider versions not being set come from this module, its hard to track that down.
So I have set the provider versions here, to clear warnings and follow good practice.

Reply:
I think it is okay for the `random` and `null` providers. For `aws` it's not good. 

Here's the long-and-drawn out explanation by our pal @antonbabenko.
 
https://medium.com/@anton.babenko/providers-version-in-terraform-modules-ab35b47a3d14